### PR TITLE
Updated 1.dqn for compatability with PyTorch 0.4 and 1.0

### PR DESCRIPTION
**Pull request by joleeson**
1. [Updated for compatibility with latest PyTorch versions.](https://pytorch.org/blog/pytorch-0_4_0-migration-guide/)  (more thorough than recommendations in #20)
- no longer uses the deprecated "Variable" class
- use of appropriate dtypes
- cpu/gpu agnostic code
- use of tensor.item() for conversion of 0-dimensional tensors to ordinary python numbers

2. Made changes such that the algorithm more closely matches that in Mnih et al. (2015) and other DQN literature:
- linear epsilon decay
- frame stacking
- training frequency is now once every 4 steps in the environment for Atari env
- option of using [Huber loss instead of RMS loss](https://blog.openai.com/openai-baselines-dqn/) in def compute_td_loss()

3. Borrowed monitoring wrapper from [OpenAI's Baselines](https://github.com/openai/baselines) to log progress of training.
4. Modified the wrappers such that it now accommodates stacked frames #9 , and outputs them as a LazyFrames object. Axes of the data is appropriately swapped for PyTorch i.e. (no. of channels)x(breadth)x(height)

### CODE REVIEW

File: 1.dqn.ipynb [hunk 1]
```diff
@@ -14,13 +14,17 @@
     "import torch\n",
     "import torch.nn as nn\n",
     "import torch.optim as optim\n",
-    "import torch.autograd as autograd \n",
-    "import torch.nn.functional as F"
+    "import torch.autograd as autograd\n",
+    "import torch.nn.functional as F\n",
+    "\n",
+    "import os\n",
+    "import logger\n",
```
Comment (colin-leu) [inline 1.dqn.ipynb:21]: which specific module is this? (can't find a module named logger)
